File: flipt-java/src/main/java/io/flipt/api/authentication/JWTAuthenticationStrategy.java
package io.flipt.api.authentication;

public final class JWTAuthenticationStrategy implements AuthenticationStrategy {
  private final String jwtToken;

  public JWTAuthenticationStrategy(String jwtToken) {
    this.jwtToken = jwtToken;
  }

  @Override
  public String getAuthorizationHeader() {
    return String.format("JWT %s", this.jwtToken);
  }
}

File: flipt-csharp/src/Authentication/JWTAuthenticationStrategy.cs
namespace Flipt.Authentication;

/// <summary>
/// This class is responsible for providing the JWT authentication strategy.
/// </summary>
public class JWTAuthenticationStrategy : IAuthenticationStrategy
{
    private readonly string _jwtToken;

    public JWTAuthenticationStrategy(string jwtToken)
    {
        _jwtToken = jwtToken;
    }

    public KeyValuePair<string, string> GetAuthorizationHeader()
    {
        return new KeyValuePair<string, string>("JWT", _jwtToken);
    }
}